hazelcast/src/test/java/com/hazelcast/osgi/HazelcastOSGiServiceTest.java

```diff
@@ -132,6 +206,64 @@ public void defaultInstanceExistWhenItIsSpecified() throws BundleException {
     }
 
     @Test
+    public void serviceCouldNotBeActivatedWhenThereIsExceptionWhileRegisteringDefaultInstance() throws BundleException {
+        String propValue = System.getProperty(HazelcastOSGiService.HAZELCAST_OSGI_START);
+        TestBundle testBundle = null;
+        try {
+            System.setProperty(HazelcastOSGiService.HAZELCAST_OSGI_START, "true");
+
+            TestBundleRegisterDeregisterListener registerDeregisterListener =
+                    new TestBundleRegisterDeregisterListener(HazelcastInstance.class.getName());
+            registerDeregisterListener.throwExceptionOnRegister = true;
+
+            testBundle = new TestBundle(registerDeregisterListener);
+            try {
+                testBundle.start();
+                fail("OSGI service could not be activated because of exception while registering default instance. " +
+                     "It is expected to get `IllegalStateException` here!");
+            } catch (IllegalStateException e) {
+                // Since bundle is not active, it is expected to get `IllegalStateException`
+            }
+        } finally {
+            if (propValue != null) {
```

Comment:
Restoring properties is done only by some tests not by all of them and the restoring properties are different from test to test. So I though that this approach is less complex.